### PR TITLE
[Ray] refine ray dataset integration

### PR DESCRIPTION
**Pull request by chaokunyang**
## What do these changes do?
Remove custom ray dataset subclasses for integration

## Check code requirements

- [ ] tests added / passed (if needed)
- [ ] Ensure all linting tests pass, see [here](https://docs.pymars.org/en/latest/development/contributing.html#check-code-styles) for how to run them

### CODE REVIEW

File: mars/dataframe/datasource/read_raydataset.py [hunk 1]
```diff
@@ -144,6 +144,10 @@ def read_raydataset(ds, columns=None, incremental_index=False, **kwargs):
     return op(index_value=index_value, columns_value=columns_value, dtypes=dtypes)
 
 
+# keep it for back compatibility
+read_raydataset = read_ray_dataset
```
Comment (qinxuye) [inline mars/dataframe/datasource/read_raydataset.py:148]: Wrap `read_raydataset` with DeprecationWarning.

**Comment by hekaisheng** [comment]
There are two issues about Ray version, could you check them in this PR?

- The limitation here seems outdated:

https://github.com/mars-project/mars/blob/df1492c4974bb027e7916ae116b12a340e3abdea/setup.cfg#L75

- The minimal version in doc(https://docs.pymars.org/en/latest/installation/ray.html#basic-steps) will raise `AttributeError` when I run `test_convert_to_ray_dataset`.

**Comment by chaokunyang** [comment]
> 

Ray Datasets is in beta in ray 1.8.0, I think we should use `ray>=1.8.0` @hekaisheng @qinxuye @fyrestone

**Comment by chaokunyang** [comment]
> 

Ray Datasets is in beta in ray 1.8.0, I think we should use ray>=1.8.0 @hekaisheng @qinxuye @fyrestone

**Comment by chaokunyang** [comment]
> There are two issues about Ray version, could you check them in this PR?
> 
> * The limitation here seems outdated:
> 
> https://github.com/mars-project/mars/blob/df1492c4974bb027e7916ae116b12a340e3abdea/setup.cfg#L75
> 
> * The minimal version in doc(https://docs.pymars.org/en/latest/installation/ray.html#basic-steps) will raise `AttributeError` when I run `test_convert_to_ray_dataset`.

Could you paste the complete error stack here?

**Comment by hekaisheng** [comment]
> > There are two issues about Ray version, could you check them in this PR?
> > 
> > * The limitation here seems outdated:
> > 
> > https://github.com/mars-project/mars/blob/df1492c4974bb027e7916ae116b12a340e3abdea/setup.cfg#L75
> > 
> > 
> > * The minimal version in doc(https://docs.pymars.org/en/latest/installation/ray.html#basic-steps) will raise `AttributeError` when I run `test_convert_to_ray_dataset`.
> 
> Could you paste the complete error stack here?

```
2022-02-15 14:16:32,951	INFO worker.py:826 -- Connecting to existing Ray cluster at address: 30.225.12.67:6379
100%|██████████| 100.0/100 [00:00<00:00, 202.73it/s]
100%|██████████| 100.0/100 [00:02<00:00, 35.20it/s]
FAILED
mars/dataframe/contrib/raydataset/tests/test_raydataset.py:54 (test_convert_to_ray_dataset[test_option0])
ray_large_cluster = None
create_cluster = <mars.deploy.oscar.ray.RayClient object at 0x7f9ca27e4d60>
test_option = [3, 3]

    @require_ray
    @pytest.mark.asyncio
    @pytest.mark.parametrize("test_option", [[3, 3], [3, 2], [None, None]])
    async def test_convert_to_ray_dataset(ray_large_cluster, create_cluster, test_option):
        assert create_cluster.session
        session = new_session(address=create_cluster.address, backend="oscar", default=True)
        with session:
            value = np.random.rand(10, 10)
            chunk_size, num_shards = test_option
            df: md.DataFrame = md.DataFrame(value, chunk_size=chunk_size)
            df.execute()
    
>           ds = mdd.to_ray_dataset(df, num_shards=num_shards)

mars/dataframe/contrib/raydataset/tests/test_raydataset.py:67: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
mars/dataframe/contrib/raydataset/dataset.py:68: in to_ray_dataset
    return _Dataset(df, ray_dataset.from_pandas_refs(chunk_refs)._blocks)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <mars.utils.lazy_import.<locals>.LazyModule object at 0x7f9ca04c6af0>
item = 'from_pandas_refs'

    def __getattr__(self, item):
        if item.startswith("_pytest") or item in ("__bases__", "__test__"):
            raise AttributeError(item)
    
        real_mod = importlib.import_module(name, package=package)
        if globals is not None and rename in globals:
            globals[rename] = real_mod
        elif locals is not None:
            locals[rename] = real_mod
>       return getattr(real_mod, item)
E       AttributeError: module 'ray.data' has no attribute 'from_pandas_refs'

mars/utils.py:332: AttributeError
```

**Comment by hekaisheng** [comment]
> > 
> 
> Ray Datasets is in beta in ray 1.8.0, I think we should use ray>=1.8.0 @hekaisheng @qinxuye @fyrestone

So the minimal version should be changed in documentation and `setup.cfg`.

**Comment by chaokunyang** [comment]
@hekaisheng @qinxuye I've addressed all the comments, could you please take a look again?

**Comment by hekaisheng** [comment]
@fyrestone Could you have a look first?

**Comment by qinxuye** [comment]
Ci cannot pass, could you please see what has happened there? @chaokunyang

**Comment by chaokunyang** [comment]
> Ci cannot pass, could you please see what has happened there? @chaokunyang

The ci failure seems not related to this pr:
```
FAILED mars/deploy/oscar/tests/test_ray_fault_injection.py::test_rerun_subtask_describe[fault_config0-fault_cluster0]
```
Could you take a loot at it? @fyrestone

**Review by fyrestone — APPROVED** [review]
LGTM

**Review by hekaisheng — APPROVED** [review]
LGTM

**Review by qinxuye — APPROVED** [review]
LGTM

**Pull request merged**